src/settings.cpp

```diff
@@ -20,11 +20,32 @@
 #include "settings.h"
 
 #include <QtCore/QSettings>
-#include <QDebug>
+#include <QStandardPaths>
+#include <QCoreApplication>
+#include <QFile>
+#include <QDir>
 
 Settings::Settings(QObject *parent) :
-    QObject(parent), m_settings(new QSettings(this))
-{
+    QObject(parent)
+{
+    const QString settings_path =
+        QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation)
+        + "/" + QCoreApplication::applicationName() + ".conf";
+
+    // Migrate old settings and delete them
+    const QString old_settings_path =
+            QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) +
+            "/harbour-quickddit/Quickddit.conf";
+    QFile old_settings_file(old_settings_path);
+    if(old_settings_file.exists()) {
+        QDir().mkpath(QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation));
+        if(old_settings_file.copy(settings_path)) {
+            old_settings_file.remove();
```

Comment:
If I remember correctly, once the new sailjail profile is in place, the old configuration directory is no longer writable. Or at least it used to be so, this might have changed. But that's the reason why the example I gave in forum had that "migrated" setting telling if we already migrated or not. Have you tested if this actually works, or do we end up getting the new settings overwritten with old ones at each startup?

 I think we could also just modify the `if` so that it also checked if the new settings file exists, and only migrate if both the old one exists and new one doesn't.

Reply:
Yeah, I tested it both with and without SailJail on SailFish OS 4.4.0.72. When SailJail is enabled, the old directory cannot be deleted, but the actual configuration file can. This seemed more elegant to me than leaving an unused configuration file that could lead to confusion.

Good point about the modified `if` though. I'll change that.